### PR TITLE
Document `VCPKG_USE_HOST_TOOLS`

### PR DESCRIPTION
**Pull request by vicroms**

### CODE REVIEW

File: vcpkg/users/triplets.md [hunk 1]
```diff
@@ -210,6 +210,20 @@ tracked as part of the package abis.
 This will cause [Binary Caching](binarycaching.md) to reuse builds from older or
 newer compilers.
 
+### VCPKG_USE_HOST_TOOLS
```
Comment (BillyONeal) [inline vcpkg/users/triplets.md:213]: I'm not sure why this is in the 'triplets' section given that I don't believe it's interpreted there.

The only mentions I see of this variable in the product are in `scripts/buildsystems/vcpkg.cmake` and triplet variables will not be defined there.

Moreover, this controls nothing about vcpkg; it controls which 'tools' directories are added to CMAKE_PROGRAM_PATH in the *user's* build system.

(Of course if someone wants to set this in their triplet they can do it through VCPKG_CMAKE_CONFIGURE_OPTIONS)

**Review by BillyONeal — APPROVED** [review]
It's not very clear to me that `VCPKG_USE_HOST_TOOLS` goes in the user's `CMakeLists.txt` from this text but I think this is better than the status quo.

**Comment by dg0yt** [comment]
IMO this option shouldn't exist. If vcpkg is a using a different host triplet, it must use that triplet for CMAKE_PROGRAM_PATH.
https://github.com/microsoft/vcpkg/pull/25529

**Comment by Neumann-A** [comment]
> IMO this option shouldn't exist. If vcpkg is a using a different host triplet, it must use that triplet for CMAKE_PROGRAM_PATH.
 
Isn't that how it works? If the host triplet is set `VCPKG_USE_HOST_TOOLS` should be true.

**Comment by dg0yt** [comment]
> Isn't that how it works? If the host triplet is set `VCPKG_USE_HOST_TOOLS` should be true.

Yes. "IF THE HOST TRIPLET IS SET".

The problem is that if the user only sets `VCPKG_TARGET_TRIPLET`, the vcpkg tool makes *independent* (and valid) decisions about the host triplet. But the user project isn't informed about them. 

https://github.com/microsoft/vcpkg/pull/25529/ solved the missing link for CMake.
https://github.com/microsoft/vcpkg/pull/25529/files#r932552438 requested the same for msbuild. But nobody contributed a proposal.

**Comment by Neumann-A** [comment]
> The problem is that if the user only sets VCPKG_TARGET_TRIPLET, the vcpkg tool makes independent (and valid) decisions about the host triplet. But the user project isn't informed about them.

That was discussed when the variables where introduce. The maintainers wanted to keep the old behavior if `VCPKG_HOST_TRIPLET` is not set. In my opinion `vcpkg.cmake` should be changed so that `VCPKG_HOST_TRIPLET` is the main variable and `VCPKG_TTARGET_TRIPLET` is the optional one.

**Comment by dg0yt** [comment]
> In my opinion `vcpkg.cmake` should be changed so that `VCPKG_HOST_TRIPLET` is the main variable and `VCPKG_TTARGET_TRIPLET` is the optional one.

I don't agree. 
It is expected that a build system can sufficiently discover the host system (`VCPKG_HOST_TRIPLET`).
It is expected that a build system must be told to build for a different target (`VCPKG_TARGET_TRIPLET`).
It is unexpected that a buildsystem needs redundant information about target and host (user project vs. integrated vcpkg)

**Comment by Neumann-A** [comment]
> It is expected that a build system can sufficiently discover the host system (VCPKG_HOST_TRIPLET).

I would agree with that if there would be one correct one and not N. There have been M issues about why vcpkg builds stuff twice if it is considered a native build. The only triplet env variable I have set is `VCPKG_DEFAULT_HOST_TRIPLET`. If it is not I would set `--host-triplet` instead of `--triplet`. 
The root problem by setting `VCPKG_TARGET_TRIPLET` is that the tool does not know if it is a cross build or not and the result thus can be both. Maybe a new variable like `VCPKG_TRIPLET` is needed to tell that it is a native build with that triplet and everything else is considered a cross build.

**Comment by dg0yt** [comment]
Let's move the focus back to `VPCKG_USE_HOST_TOOLS`. The key point is that this doesn't need to be documented or set by the user. The vcpkg toolchain must ensure a match between toolchain configuration and vcpkg tool decisions.

File: vcpkg/users/host-dependencies.md [hunk 2]
```diff
@@ -73,3 +73,19 @@ tool ports are common examples. In this case, you can use the `"native"`
...
 HOST_TRIPLET`).
+
+## VCPKG_USE_HOST_TOOLS
+
+Set this CMake option to add executables built by vcpkg to
+[`CMAKE_PROGRAM_PATH`](https://cmake.org/cmake/help/latest/variable/CMAKE_PROGRAM_PATH.html).
+and make them available during your build process.
+
+For this option to take effect it must be set before any call to `project()` in
+your `CMakeLists.txt` file and the `VCPKG_HOST_TRIPLET` variable must also be set.
```
Comment (AugP) [inline vcpkg/users/host-dependencies.md:84]:
```suggestion
For this option to take effect it must be set before any call to `project()` in
your `CMakeLists.txt` file. The `VCPKG_HOST_TRIPLET` variable must also be set.
```

**Pull request merged**